parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -36,19 +36,33 @@ configureTransparentHugePage() {
 
 {{- if ShouldConfigSwapFile}}
 configureSwapFile() {
-    SWAP_SIZE_KB=$(expr {{GetSwapFileSizeMB}} \* 1000)
-    DISK_FREE_KB=$(df /dev/sdb1 | sed 1d | awk '{print $4}')
-    if [[ ${DISK_FREE_KB} -gt ${SWAP_SIZE_KB} ]]; then
-        SWAP_LOCATION=/mnt/swapfile
-        retrycmd_if_failure 24 5 25 fallocate -l ${SWAP_SIZE_KB}K ${SWAP_LOCATION} || exit $ERR_SWAP_CREAT_FAIL
-        chmod 600 ${SWAP_LOCATION}
-        retrycmd_if_failure 24 5 25 mkswap ${SWAP_LOCATION} || exit $ERR_SWAP_CREAT_FAIL
-        retrycmd_if_failure 24 5 25 swapon ${SWAP_LOCATION} || exit $ERR_SWAP_CREAT_FAIL
-        retrycmd_if_failure 24 5 25 swapon --show | grep ${SWAP_LOCATION} || exit $ERR_SWAP_CREAT_FAIL
-        echo "${SWAP_LOCATION} none swap sw 0 0" >> /etc/fstab
+    # https://learn.microsoft.com/en-us/troubleshoot/azure/virtual-machines/troubleshoot-device-names-problems#identify-disk-luns
+    swap_size_kb=$(expr {{GetSwapFileSizeMB}} \* 1000)
+
+    if [[ -L /dev/disk/azure/resource-part1 ]]; then
+        echo "Will use resource disk for swap file"
+        disk_id=$(readlink /dev/disk/azure/resource-part1)
```

Comment:
annoyingly I think we actually have to check both for space with our existing behavior, right?
```
if temp disk
  if has_space:
     use it
  else
     if os disk has space
       use it
  else
    hope this never happens but I guess possible with 30GB os disk if you ask for swap -> TODO fail in RP
```

Reply:
yep yep, this new behavior first resolves the device, then checks space and acts accordingly

Reply:
yep, good catch